python/graphstorm/model/rgcn_encoder.py

```diff
@@ -168,8 +168,9 @@ def forward(self, g, inputs):
                     # tensor. This tensor won't be used for computing embeddings.
                     # We need this just to fulfill the requirements of DGL message passing
                     # modules.
-                    assert not self.self_loop, \
-                            f"We cannot allow self-loop if node {k} doesn't have input features."
+                    if g.num_dst_nodes(k) > 0:
```

Comment:
In which case g.num_dst_nodes(k) == 0?
If dst node is zero, why there is an MFG?

Reply:
I'm not sure why. I got this issue when running training on the mag dataset.

Reply:
If we remove this assertion, will there be any potential error?